src/recipe.rs

```diff
@@ -16,6 +16,76 @@ fn error_from_signal(recipe: &str, line_number: Option<usize>, exit_status: Exit
   }
 }
 
+#[derive(Debug)]
+struct SystemMap {
+  windows: bool,
+  macos: bool,
+  linux: bool,
+  openbsd: bool,
+  unix: bool,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum System {
+  Windows,
+  MacOS,
+  Linux,
+  OpenBSD,
+  Unix,
+}
+
+impl System {
+  fn current() -> System {
+    use System::*;
+    if cfg!(target_os = "linux") {
+      return Linux;
+    }
+    if cfg!(target_os = "openbsd") {
+      return OpenBSD;
+    }
+    if cfg!(target_os = "macos") {
+      return MacOS;
+    }
+    if cfg!(target_os = "windows") || cfg!(windows) {
+      return Windows;
+    }
+    if cfg!(unix) {
+      return Unix;
+    }
+    panic!("No recognized system");
+  }
+
+  fn enabled(self, enabled: SystemMap, disabled: SystemMap) -> bool {
```

Comment:
I think we need some kind of more extensive testing for this. It looks right to me, but it strikes me as very fragile, and easy to get wrong when extending..

Reply:
Yeah, makes senses - I'm not not entirely happy with the fragility of this code either, but I don't have any ideas off the top of my head for how to make it more robust. Do you have any ideas for how the tests should be architected?

Reply:
One idea:

```
#[test]
fn foo() {
  fn case(attributes: &str, active: bool) {
    Test::new().justfile(format!("[{attributes}]\nfoo:"))
    // assert that foo is or isn't active, depending on attributes.
  }

  case("macos",  cfg!(target_os = "darwin"));
}

One idea for improving the structure would be to try to factor out this pattern:

```
      System::Windows => {
        !disabled.windows
          && (enabled.windows
            || !(enabled.macos || enabled.linux || enabled.openbsd || enabled.unix))
      }
```


Maybe with maps:

```
        !disabled[self] && (enabled[self] || !self.others().iter().any(|other| system.enabled()))
```